### DISCUSSION TITLE
How to import .css file on a per page/component basis?

### DISCUSSION THREAD

**Discussion by chhuang** [vercel/next.js#10835] [Help]
I understand that we can import global CSS in `_app` and the built-in CSS module support. 

But neither of these approaches solve my problem: I'm using a third party image gallery and it requires to import a `.css` file, but it's not good practice to import it in `_app` because the gallery is only used on one page. What's the best way to import it just for that page?

Related to #10665

**Comment by waptik** [comment]
Did you try this in your components or pages: eg:
`import  'PATH_TO_CSS_LOCATION/styles.css'` ?

**Reply by chhuang** [reply]
Yes I did:

```
Global CSS cannot be imported from files other than your Custom <App>. Please move all global CSS imports to pages/_app.js.
```

**Reply by waptik** [reply]
I don't know your configuration or nextjs version but I think this might help.
https://github.com/opencollective/opencollective-frontend/blob/dd6213172bf88272c3013b32da8ef1456bd6ba50/pages/marketingPage.js
Go through their configuration and see how you can solve your issue

**Reply by chhuang** [reply]
Thanks for the link, I'll definitely try that. I'm using `next@latest`.

**Reply by waptik** [reply]
Hey! You can check this out https://github.com/zeit/next.js/discussions/10849#discussioncomment-1022 @chhuang

**Reply by chhuang** [reply]
Thanks but this wouldn't work for my use case because it applies the CSS on a per class basis, which is good for adding component level CSS. What I really want to do is slightly different, which is adding/importing a global predefined CSS file within a component.

**Comment by gragland** [comment]
If you use the [next-css](https://github.com/zeit/next-plugins/tree/master/packages/next-css) plugin then it should disable their new CSS import rules.

**Comment by ebs-PriyankaKanwar** [comment]
we have updated next JS, all things are working fine except one the CSS is not loaded in proper order in production mode, however, it's working fine in dev mode.

Here are the versions
Node: v13.7.0
React:^16.12.0
NextJs:9.1.6
OS: Linux - ubuntu

Here is what I got when did build:

├ static/pages/_app.js 16.1 kB
├ chunks/1899c3d08f1d4a9ab6e4e22050e66cd1600464e8.b9ea19.js 3.53 kB
├ chunks/48183758.ede116.js 65 B
├ chunks/57a5805a715b88beb8aec3ac43d8f962ef8879cc.53c4e4.js 2.34 kB
├ chunks/9ed67fcaced240be487e7a2880bfff868999eacb.76e83a.js 8.29 kB
├ chunks/ab68c4ca07cacf4e3048ee2d6f69810106b4b489.68a8b3.js 10.4 kB
├ chunks/d32b6132cd0996134c9f87f7274f553f791cd563.18dde5.js 6.39 kB
├ chunks/d6e4d54ae3ad4e326943499c7720eb830c61721b.b7693f.js 4.41 kB
├ chunks/framework.4dd100.js 40.3 kB
├ runtime/main.d3f0e8.js 13.2 kB
├ runtime/webpack.361a6a.js 1.59 kB
├ css/16c4518f007189356065.css 54.9 kB
└ css/f93669f8200e9712e425.css 30.6 kB

Here is my next.config :
const withFonts = require('next-fonts');
const withImages = require('next-images');
// const withCSS = require('@zeit/next-css')

const withPlugins = require('next-compose-plugins');
const nextEnv = require('next-env');
const dotenvLoad = require('dotenv-load');

dotenvLoad();

withFonts({
webpack(config, options) {
// config.plugins.push(new webpack.EnvironmentPlugin(localEnv))
return config;
}
})

module.exports = withPlugins([

nextEnv({
staticPrefix: 'CUSTOM_STATIC_',
publicPrefix: 'CUSTOM_PUBLIC_',
}),withFonts, withImages
]);

I have two things here
1 - why we are having two files of CSS when created build? How can we merge into one CSS file?
2 - How I can fix it so that it will work in the same manner as in dev mode?

Any suggestions will be helpful.

**Comment by jasonwilliams** [comment]
The next-css plugin shouldn't be described as `legacy` until the built-in css has parity with its feature set. 
In this case its not possible to import css on a page-by page basis, so i'm sure many of us will need to continue using next-css.

Should there be a warning in the terminal whilst this is still the case?
Calling it legacy signals that we should be moving to the built-in solution but that's misleading as its not a direct replacement to next-css.
